src/renderers/shared/fiber/__tests__/ReactIncrementalTriangle-test.js

```diff
@@ -0,0 +1,325 @@
+    let keyCounter = 0;
+    function reset(nextStep = 0) {
+      triangles = [];
+      leafTriangles = [];
+      // Remounts the whole tree by changing the key
+      ReactNoop.render(<App depth={depth} key={keyCounter++} />);
+      ReactNoop.flush();
+      treeIsConsistent();
+      return appInstance;
+    }
+
+    reset();
+    const totalChildren = leafTriangles.length;
+    const totalTriangles = triangles.length;
+
+    function treeIsConsistent(activeTriangle, counter) {
+      const activeIndex = activeTriangle ? activeTriangle.leafIndex : -1;
+
+      const children = ReactNoop.getChildren();
+      for (let i = 0; i < children.length; i++) {
+        let child = children[i];
+        let num = child.prop;
+
+        // If an expected counter is not specified, use the value of the
+        // first child.
+        if (counter === undefined) {
+          if (typeof num === 'string') {
+            counter = num.substr(1, num.length - 2);
+          } else {
+            counter = num;
+          }
+        }
+
+        if (i === activeIndex) {
+          if (num !== `*${counter}*`) {
+            throw new Error(
+              `Triangle ${i} is inconsistent: ${num} instead of *${counter}*.`,
+            );
+          }
+        } else {
+          if (num !== counter) {
+            throw new Error(
+              `Triangle ${i} is inconsistent: ${num} instead of ${counter}.`,
+            );
+          }
+        }
+      }
+    }
+
+    function simulate(...actions) {
+      const app = reset();
+      let expectedCounterAtEnd = app.state.counter;
+
+      let activeTriangle = null;
+      for (let i = 0; i < actions.length; i++) {
+        const action = actions[i];
+        switch (action.type) {
+          case FLUSH:
+            ReactNoop.flushUnitsOfWork(action.unitsOfWork);
+            break;
+          case STEP:
+            app.setCounter(action.counter);
+            expectedCounterAtEnd = action.counter;
+            break;
+          case INTERRUPT:
+            app.interrupt();
+            break;
+          case TOGGLE:
+            const targetTriangle = leafTriangles[action.childIndex];
+            if (targetTriangle === undefined) {
+              throw new Error('Target index is out of bounds');
+            }
+            if (targetTriangle === activeTriangle) {
+              activeTriangle = null;
+              targetTriangle.deactivate();
+            } else {
+              if (activeTriangle !== null) {
+                activeTriangle.deactivate();
+              }
+              activeTriangle = targetTriangle;
+              targetTriangle.activate();
+            }
+            ReactNoop.flushAnimationPri();
+            break;
+          default:
+            break;
+        }
+      }
+      // Flush remaining work
+      ReactNoop.flush();
+      treeIsConsistent(activeTriangle, expectedCounterAtEnd);
+    }
+
+    return {simulate, totalChildren, totalTriangles};
+  }
+
+  it('works', () => {
```

Comment:
We could also `xit` them for now to skip, and when we fix them, immediately `it` them so they don't regress. This would be almost equivalent, except that it wouldn't get indication when a skipped test suddenly unintentionally passes (since you have to explicit `it` it to see if it passed).